### DISCUSSION TITLE
Improve read/write splitting in CP by using replicas for element-indexes

### DISCUSSION THREAD

**Discussion by FlorisDerks** [craftcms/cms#9996] [Ideas] [answered]
Since Craft CMS 3.5.11.1 all POST-requests in the CP explicitly disable any configured replicas from being used for any queries. I suspect this is done for data integrity, mainly because the save-functions store data and then immediately re-query it to fill the newly created model.

However, a side-effect of this change is that all element-indexes exclusively use the writer to query their entries, because `/index.php?p=admin/actions/element-indexes/get-elements` and `/index.php?p=admin/actions/element-indexes/count-elements` use POST. Because those queries could be quite heavy (an element-index takes 6-12 seconds to load in my case due to having a lot of data) it creates a risk for the writer-instance to be overwhelmed with SELECT-queries that could easily be handled by a replica.

Would it be possible to disable replicas in a more fine-grained way? For example by only disabling them when a transaction is started (and re-enabling them after)? Or at least by explicitly allowing them for element-indexes?

**Comment by brandonkelly** [comment, marked as answer]
Yeah good call. Just made that change for the next release (6b2531f590061039e3f232a7c8dfd7efd83a7e76).

**Reply by brandonkelly** [reply]
Craft 3.7.18 is out now with that change.
